### ISSUE TITLE
StackOverflowError

### ISSUE THREAD

**Issue by James137137** [TheCricket/FTB-Islands#17]
at https://github.com/TheCricket/FTB-Islands/blob/master/src/main/java/com/cricketcraft/ftbisland/util/IslandCreator.java#L164
 there is a StackOverflowError

**Comment by James137137** [comment]
I suggest using something like jsonSerializer = (new GsonBuilder()).setPrettyPrinting().registerTypeAdapter(IslandPos.class, this).create(); in your intizing  part

**Comment by TheCricket** [comment]
We know :) Thats kind of why you shouldn't build the mod from github and try to use it

**Comment by James137137** [comment]
I know. I've already converted it to a bukkit plugin and fixed all the bugs XD (still in progress with some of the stuff)

**Comment by TheCricket** [comment]
No that is not allowed

**Comment by TheCricket** [comment]
I ask you take that down immediately wherever it is hosted

**Comment by James137137** [comment]
ok. deleted from test server (on personally computer)

**Comment by James137137** [comment]
ok may I use a plugin to hook into the mod regarding the term: "Adapted Material such as addon mods is encouraged" ?

**Comment by TheCricket** [comment]
Don't do that to people without asking permission

**Comment by James137137** [comment]
no problem. 

so regarding to Section 2 c: (Adapted Material such as addon mods is encouraged. All other works should be discussed with the Owner) may I have permission create a plugin/mod to do the following?

Change the .ser database to .json (so its easier to edit)?
Make it so anyone can /island spawn not just OP (makes it more automatic for the server rather than command signs)?
add a /island recreate method?

**Comment by kolatra** [comment]
We're rewriting a lot of the mod, and json is one thing that will happen for sure.

**Comment by TheCricket** [comment]
No you may not make a plugin

Reason being is I will not be maintaining it and you will be shipping it as a standalone. This will cause people to report issues which are not mine to me and I will not have that. I left the modded community because of this reason.

If you want to submit a pull request you may.

**Comment by sameer** [comment]
Lol!

**Comment by sameer** [comment]
I'm not sure what you expect to happen by posting your code here on GitHub....or by even writing Java code in general. Anyone can reverse engineer code written in Java...that is why Minecraft Forge and Bukkit exist in the first place. If you really want to be fancy, maybe you should've written it in C/C++ with Java Native Integration.
- How are you going to stop someone from using your mod on their server? You clearly have no way of forcing them to do that. Sure, you could ask them to remove it, but what if they don't listen to you and instead ban you? If you bypass the ban and continue to harass them without any actual proof, it's harassment. If that person is under 18, you are therefore harassing a minor and could be asked to cease and desist.
- How are you going to stop someone from recreating your mod and making it better? If people want to do this, you should invite them to help you fix it, not tell them "no please don't do that".
- How do you know @James137137 actually deleted his adapted version? It probably took him a lot of time to write that, and I doubt he just up and deleted it because you threatened him. You don't have the right to dictate what files can/cannot be on another person's computer....you don't even know who he really is.

Honestly, this is pretty childish and immature behavior on your part as a mod developer. Shame on you! After seeing all of this, I hope someone reports you to the FTB Team and has your mods removed!

**Comment by jellysquid3** [comment]
While I'm not a part of this, by hosting your source code publicly on GitHub [you're agreeing to](https://help.github.com/articles/github-terms-of-service/) giving people the ability to fork and modify your code (essentially allowing re-uploading of source code as well).

If you don't want that, you shouldn't be hosting a public repo on GitHub where users expect those basic rights.

**Comment by NightDevel** [comment]
I am compiling your plugin here:
https://ci.aperture.lol/browse/THECRICKET-FTBISLANDS

I am also mirroring your entire github user here for my personal use: 
https://git.aperture.lol/projects/THECRICKET

**Comment by sameer** [comment]
Inb4 TheCricket deletes all of this because it's bad publicity, let me screenshot it and save the page.

**Comment by NightDevel** [comment]
That would be a darn shame...
hmm...

https://git.aperture.lol/pages/THECRICKET/archive-comments/master/browse/github.com/TheCricket/FTB-Islands/issues/17.html

**Comment by FaMatrix** [comment]
Already forwarded thread information to FTB Team, I don't think it would work that way, but.. +Kudos to @Robotia and @nightDevel..!

**Comment by sameer** [comment]
@Azurity-y Thank you for helping end cancerous behavior like this. :+1:

**Comment by Yive** [comment]
Probably gonna get a lot of down votes for this, but that doesn't matter to me.

I am sorry to both you @TheCricket & even @Kolatra, this has gone too far with @Robotia's unprofessional remarks. I will be detaching his project from my github group. I understand your reasoning as to what you're saying, but I wont refrain from stating my opinion on your choices since this is your mod & you should be allowed to do what you want with it.

I am deeply sorry for all the backlash you've gotten today & yesterday from your choices. I hope for your mod to still be accepted even after it was reported to the FTB Team. It's a shame people have taken this so far when you were just making your own choices for what you want for your mod.

PS: I know I wasn't involved, but I feel like I should write this comment since a lot of conversation about this issue was talked about in both of my public chats.

**Comment by sameer** [comment]
I'd like to apologize as well, I went too far with this.
I would just like to point out that, as @Collin1971 and @NightDevel have shown, you are trying to restrict access to your code on a public repository, which is not really a feasible endeavor. The mod should've been made closed-source in the first place if you did not want anyone to build it independently or try to make modifications without a PR.

Just as a final remark, I would like to say that the attitude TheCricket was showing is not uncommon. It is very discouraging to people who are just entering Java development and portrays a false view of the Minecraft mod development community. There are plenty of people out there who are very nice about the mods they develop and like to solve issues, but what I saw here (https://github.com/TheCricket/FTB-Islands/pull/19) did not show that.

Thank you to whoever reads this, I hope that the community can work on this mod to fix pressing issues, like the use of the ObjectOutputStream for data storage.

Edit: I thought that @Kolatra was being very nice about the issue, just not TheCricket.
